tests/test_airpyllution.py

```diff
@@ -1,55 +1,84 @@
 from airpyllution import airpyllution
 import pandas as pd
+from pandas._testing import assert_frame_equal
+from unittest.mock import patch
+from constants import *
+from airpyllution.utils import *
 
-def test_pollution_history():
-    """Test word counting from a file."""
-    mock_params = {
-        'lat': 49.28,
-        'lon': 123.12,
-        'start': 1606488670,
-        'end': 1606747870,
-        'appid': 'mock_api_key'
-    }
-
-    mock_incorrect_params = {
-        'lat': 'latitude_val',
-        'lon': 'longitude_val',
-        'start': 1234.567,
-        'end': 3.14159,
-        'appid': 'fake'
-    }
 
+def mocked_requests_get_pollution_history(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+    print(kwargs)
+    if kwargs['url'] == 'http://api.openweathermap.org/data/2.5/air_pollution/history':
+
+        if kwargs['params']['appid'] == 'invalid_api_key':
+            return MockResponse(mock_api_invalid_key_error, 404)
+
+        return MockResponse(mock_history_data, 200)
+
+    elif kwargs['url'] == 'http://api.openweathermap.org/data/2.5/air_pollution/forecast':
+        return MockResponse(mock_forecast_data, 200)
```

Comment:
@christopheralex You'll need to use the `@patch` decorator on top of your test function to have access to the `MockResponse` object. 
You can then add `mock_forecast_data` to the constants file which will be used instead of the actual data from the API.

Reply:
Yup. But looks like we have merge conflict between 2 PRs.

Reply:
Okay on it .